namespace missing from build.grandle

> Namespace not specified. Specify a namespace in the module's build file: C:\Users\Admin\AppData\Local\Pub\Cache\hosted\pub.dev\audiotags-1.4.0\android\build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

I'm having the same problem!

I'm on version 8.5.1 of AGP (Android Gradle Plugin) and Gradle is on version 8.7-all. 
I'm compiling for android with targetSdk 34 and minSdk 22 under flutter version 3.22.2 and dart 3.4.3 stable.

```bash
Launching lib\main.dart on sdk gphone64 x86 64 in debug mode...

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':audiotags'.
> Could not create an instance of type com.android.build.api.variant.impl.LibraryVariantBuilderImpl.
   > Namespace not specified. Specify a namespace in the module's build file: C:\Users\joaol\AppData\Local\Pub\Cache\hosted\pub.dev\audiotags-1.4.0\android\build.gradle. See https://d.android.com/r/tools/upgrade-assistant/set-namespace for information about setting the namespace.

     If you've specified the package attribute in the source AndroidManifest.xml, you can use the AGP Upgrade Assistant to migrate to the namespace value in the build file. Refer to https://d.android.com/r/tools/upgrade-assistant/agp-upgrade-assistant for general information about using the AGP Upgrade Assistant.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2s
Error: Gradle task assembleDebug failed with exit code 1

Exited (1).
```

![image](https://github.com/user-attachments/assets/fbbc93e0-0ada-4523-8399-7eb01a3edc8c)

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -25,6 +25,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
+    namespace = "com.erikas.audiotags"
     compileSdkVersion 31
 
     compileOptions {
@@ -42,7 +43,7 @@ android {
     }
 
     defaultConfig {
-        minSdkVersion 16
+        minSdkVersion 21
     }
 
     externalNativeBuild {

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -25,7 +25,11 @@ apply plugin: 'com.android.library'
 apply plugin: 'kotlin-android'
 
 android {
-    namespace = "com.erikas.audiotags"
+    // Conditional for compatibility with AGP < 4.2.
+    if (project.android.hasProperty("namespace")) {
+        namespace 'com.erikas.audiotags'
+    }
+
     compileSdkVersion 31
 
     compileOptions {

--- a/android/src/main/AndroidManifest.xml
+++ b/android/src/main/AndroidManifest.xml
@@ -1,2 +1,3 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+  package="com.erikas.audiotags">
 </manifest>

--- a/android/src/main/AndroidManifest.xml
+++ b/android/src/main/AndroidManifest.xml
@@ -1,3 +1,2 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-  package="com.erikas.audiotags">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 </manifest>

